php-packages/testing/src/integration/Extension/ExtensionManagerIncludeCurrent.php

```diff
@@ -112,4 +104,32 @@ protected function getAssetsFilesystem(): Cloud
     {
         return new FilesystemAdapter(new FlysystemFilesystem(new Local($this->paths->public.'/assets'), ['url' => resolve('flarum.config')->url().'/assets']));
     }
+
+    protected function includeCurrentExtension($package, string $packagePath)
+    {
+        if (Arr::get($package, 'type') === 'flarum-extension') {
+            $current = new Extension($packagePath, $package);
+            $current->setInstalled(true);
+            $current->setVersion(Arr::get($package, 'version'));
+            $current->calculateDependencies([], []);
+
+            $this->extensions->put($current->getId(), $current);
+
+            $this->extensions = $this->extensions->sortBy(function ($extension, $name) {
+                return $extension->composerJsonAttribute('extra.flarum-extension.title');
+            });
+        }
+    }
+
+    /**
+     * Allows symlinking the vendor directory in extensions when running tests on monorepos.
+     */
+    protected function includeMonorepoExtensions($package, string $packagePath)
+    {
+        foreach ($this->subExtensionConfsFromJson($package, $packagePath) ?? [] as $path => $package) {
+            $extension = $this->extensionFromJson($package, $path);
+            $extension->calculateDependencies([], []);
+            $this->extensions->put($extension->getId(), $extension);
+        }
+    }
```

Comment:
I don't love these methods as mutations of `this->extensions`; why not return an array instead?

Reply:
imo it's cleaner to mutate, since we have to mutate it after the methods anyway, the constant pass and return collection object reads oddly.